Rakefile

```diff
@@ -12,28 +12,44 @@ begin
   require 'semverse'
 
   class FakePuppetfile
-    attr_reader :new_content
-
     def initialize
       @new_content = []
     end
 
     def forge(url)
-      @new_content << "forge '#{url}'"
-      @new_content << ''
+      @new_content << ['forge', url, nil]
       PuppetForge.host = url
     end
 
     def mod(name, options = nil)
-      if options.nil?
-        @new_content << "mod '#{name}'"
-      elsif options.is_a?(String)
-        @new_content << "mod '#{name}', '#{options}'"
-      else
+      if options.is_a?(Hash) && !options.include?(:ref)
         release = PuppetForge::Module.find(name.tr('/', '-')).current_release
         version = Semverse::Version.new(release.version)
         max = "#{version.major}.#{version.minor + 1}.0"
-        @new_content << "mod '#{name}', '>= #{version} < #{max}'"
+        @new_content << ['mod', name, ">= #{version} < #{max}"]
+      else
+        @new_content << ['mod', name, options]
+      end
+    end
+
+    def content
+      max_length = @new_content.select { |type, value| type == 'mod' }.map { |type, value| value.length }.max
+
+      @new_content.each do |type, value, options|
+        if type == 'forge'
```

Comment:
is this code branch really worth it, or should be just have `"forge '#{url}'"` as a `header` variable and print that before looping over `content`?

Reply:
I implemented this in https://github.com/Katello/katello-installer/commit/0a42ab6b285613a964dd9071f70b89b5630a4691 and now just copied it. Technically you're right that it could be simplified.

Reply:
But no reason to block the PR on it :)